block_manipulation_service/src/main/java/com/stnetix/ariaddna/blockmanipulation/BlockGenerate.java

```diff
@@ -0,0 +1,63 @@
+package com.stnetix.ariaddna.blockmanipulation;
+
+import org.springframework.beans.factory.annotation.Autowired;
+
+import java.io.*;
+import java.nio.ByteBuffer;
+import java.nio.channels.FileChannel;
+import java.util.Date;
+import java.util.HashMap;
+
+/**
+ * Created by LugovoyAV on 15.02.2018.
+ */
+public class BlockGenerate {
+    private HashMap<File,Long> files = new HashMap<>();
+    private final int blockSize = 5242880;
+    @Autowired
+    private LocalService localService;
+
+    public Block getNextBlock(Metafile metaFile)
+    {
+        return createNextBlock(metaFile.getVersion(),metaFile.getVersion());
+    }
+
+    //TODO for directory
+    private Block createNextBlock(String version, String fileUuid)  {
+        long offset = 0;
+        File localFile = localService.getLocalFileByUuid(fileUuid);
+        if (!localFile.isFile())
+            return null;
+
+        if (!files.containsKey(localFile))
+            files.put(localFile,Long.valueOf(offset));
+        else
+            offset = files.get(localFile);
+
+        ByteBuffer buf = ByteBuffer.allocate(blockSize);
```

Comment:
In this algorithm we use read by blocksize and not read last block if it block less blocksize. I'm right?